public/app/plugins/datasource/elasticsearch/datasource.test.ts

```diff
@@ -58,73 +72,68 @@ const createTimeRange = (from: DateTime, to: DateTime): TimeRange => ({
   },
 });
 
-interface Args {
-  data?: any;
+interface TestContext {
+  data?: Data;
   from?: string;
-  jsonData?: any;
+  jsonData?: Partial<ElasticsearchOptions>;
   database?: string;
-  mockImplementation?: (options: BackendSrvRequest) => Observable<FetchResponse>;
+  fetchImplementation?: (options: BackendSrvRequest) => Observable<FetchResponse>;
+}
+
+interface Data {
+  [key: string]: undefined | string | string[] | number | Data | Data[];
 }
 
 function getTestContext({
-  data = {},
+  data = { responses: [] },
   from = 'now-5m',
-  jsonData = {},
-  database = '[asd-]YYYY.MM.DD',
-  mockImplementation = undefined,
-}: Args = {}) {
-  jest.clearAllMocks();
-
+  jsonData,
+  database = '[test-]YYYY.MM.DD',
+  fetchImplementation = undefined,
```

Comment:
Just a very nit thing, but what if we just combine the old and new name and name it `fetchMock`? I think `mockImplementation` states good, that this method will be a mock, but not what it will mock. `fetchImplementation` states it will implement `fetch` which is also good information, but not that this will be used to mock the original implementation.

As said, just a very nit thing.

Reply:
I don't mind this change, but `fetchMock` is already being defined later and returned from this function.

Reply:
Thanks for the change. Think this way is also fine 👍